src/PostgREST/DbStructure/Proc.hs

```diff
@@ -77,22 +79,31 @@ type ProcsMap = M.HashMap QualifiedIdentifier [ProcDescription]
   An overloaded function can have a different volatility or even a different return type.
   Ideally, handling overloaded functions should be left to pg itself. But we need to know certain proc attributes in advance.
 -}
-findProc :: QualifiedIdentifier -> S.Set Text -> Bool -> ProcsMap -> ProcDescription
-findProc qi payloadKeys paramsAsSingleObject allProcs = fromMaybe fallback bestMatch
+findProc :: QualifiedIdentifier -> S.Set Text -> Bool -> ProcsMap -> (NonEmpty ProcDescription, Bool)
```

Comment:
That return type seems strange.

Can we not just return a list of `ProcDescription` and then later do:
- if length = 0 -> not found error
- if length = 1 -> all good
- if length > 1 -> duplicate error

Reply:
It's a problem related with the `fallback` function that we commented before. There are some complications that I need to look into with more detail. After I  finish the argument type casting and the tests, I'll refactor this part of the code as you mentioned.